### PR TITLE
Add ADR-13, multi cluster environments.

### LINKED ISSUE #306: Reorganize overlays to reflect multicluster environments
Part of: https://github.com/operate-first/SRE/issues/108

Reconsider the "overlay" meaning from "a cluster" to "an environment", since we have multi cluster environments and we want to keep things clean and understandable

- [x] Merge `overlays/quicklab` and `overlays/crc` into `overlays/dev` since they are always the same ([discussion](https://github.com/operate-first/blueprint/issues/28))
- [x] Move `overlays/moc`, `overlays/moc-infra`, `overlays/moc-base` into `overlays/moc/*` as:
  - `overlays/moc/zero`
  - `overlays/moc/infra`
  - `overlays/moc/common`
- [x] ~Adjust scripts~ (No need for that)
- [x] Update documentation
- [ ] Add ADR explaning the reasoning

### PR DESCRIPTION
**Pull request by HumairAK**
Resolves https://github.com/operate-first/apps/issues/306

### CODE REVIEW

**Comment by HumairAK** [comment]
@tumido  @4n4nd @hemajv

**Review by tumido — COMMENTED** [review]
/lgtm

File: docs/adr/0013-multicluster-environments.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Cluster deployment methods
+Technical Story: [issue-1](https://github.com/operate-first/apps/issues/306)
+
+## Context and Problem Statement
+From the onset we have used the terms "environment" and "cluster" interchangeably, and our repository structure has been reflective of that. However, there seems to be a clear need for a distinction in certain contexts between environments like "MOC" and "Quicklab" and specific clusters belonging to those environments. For example, before we used to use "MOC" to refer to the original `cnv` cluster, but we now have two clusters that belong to the MOC environment (`infra` and `zero`). Furthermore, since there is no such thing as one defacto "Quicklab/CRC" cluster, when we design for these environments we have to design for a very generic environment without adding configurations tuned for one specific cluster. In consideration of these different factors, we must design with _Multi-Cluster Environments_ in mind and restructure existing repos accordingly.
+
+
+## Considered Options
+
+###1. Organize repos at Cluster level
```
Comment (4n4nd) [inline docs/adr/0013-multicluster-environments.md]: looks like there are md formatting issues

**Review by 4n4nd — COMMENTED** [review]
/lgtm

**Pull request merged**